src/validation/validator.rs

```diff
@@ -191,17 +196,27 @@ impl<
     pub fn validate(
         consignment: &'consignment C,
         resolver: &'resolver R,
-        testnet: bool,
+        chain_net: ChainNet,
         context: S::Context<'_>,
+        safe_height: Option<NonZeroU32>,
+        trusted_op_seals: BTreeSet<OpId>,
```

Comment:
This introduces a backdoor into the RGB consensus.

This method is called only externally, exposing an access to the consensus verification for the application level - which is used on top to hide the double-spent issue also introduced in the consensus (see my next comment)